csharp/Adapter/Microsoft.Spark.CSharp/Network/SocketStream.cs

```diff
@@ -124,7 +124,13 @@ public override int ReadByte()
                 recvDataCache = streamSocket.Receive();
             }
 
-            return recvDataCache.ReadByte();
+            var v = recvDataCache.ReadByte();
+            if (recvDataCache.IsReadable()) return v;
+
+            // Release cache if it is not readable.
```

Comment:
add some more context here as code comments to explain when it might happen and why we are not throwing exception etc.